File: Recursion/Python/factorial.py
def findfactorialRecursive(number):

    if number == 0:
        return 1

    return number * findfactorialRecursive(number - 1)

def findfactorialIterative(number):

    result = 1
    for i in range(1, number + 1):
        result *= i

    return result

if __name__ == '__main__':
    number = 4
    result = findfactorialRecursive(number)
    print(f"The factorial of {number} is {result}")

    result = findfactorialIterative(number)
    print(f"The factorial of {number} is {result}")

File: Recursion/C++/factorial.cpp
#include <iostream>

int findfactorialRecursive(int number) {
    if (number == 0) {
        return 1;
    }
    return number * findfactorialRecursive(number - 1);
}

int findfactorialIterative(int number) {
    int result = 1;
    for (int i = 1; i <= number; i++) {
        result *= i;
    }
    return result;
}

int main() {
    int number=5;
    int result;

    result = findfactorialRecursive(number);
    std::cout << "The factorial of " << number << " is " << result << std::endl;

    result = findfactorialIterative(number);
    std::cout << "The factorial of " << number << " is " << result << std::endl;
    return 0;
}
